Practice/Square/Program.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Square
+{
+    class GeoFigure
+    {
+        public string Name { get; private set; }
+        public double MainParametr { get; private set; }
+
+        public GeoFigure(string name, double param)
+        {
+            Name = name;
+            if (param >= 1 && param <= 10) MainParametr = param;
+
+            else Console.WriteLine("the value is wrong");
+
+        }
+
+        public virtual void ShowResult()
+        {
+            Console.Write($"This is {Name}. It's type is {this.GetType()}. It's squere is ");
+        }
+    }
+
+    class Circle : GeoFigure
+    {
+        public Circle(string name, double radius) : base(name, radius) { }
+        public override void ShowResult()
+        {
+            base.ShowResult();
+            Console.WriteLine(Math.PI * MainParametr * MainParametr);
```

Comment:
this method does two different actions:

1. Show Info about figure
2. Calc figure's square

Just read about first `SOLID` principle. `S` - single responsibility principle.